pkg/api/bugResolver.go

```diff
@@ -0,0 +1,117 @@
+package api
+
+import (
+	"encoding/json"
+	"log"
+	//"time"
+
+	//"github.com/graph-gophers/graphql-go"
+
+	"github.com/thrasher-redhat/internal-tools/pkg/bugzilla"
+)
+
+/*
+how do we do this...are we going to hold a bug in memory?
+where do we actually get the bug?
+does the Bugs function that returns a bugResolver query the DB and create one?
+why do we need all these getter functions? Could we just make bug -> Bug
+
+Resolver will do a db query and create a bugResolver (or list of them) with the that bug
+
+*/
+
+type BugResolver struct {
+	bug bugzilla.Bug
+}
+
+// Resolvers
+
+/* For Reference...
+    id: Int!
+    datestamp: String!
+
+    component: String!
+    status: String!
+    summary: String!
+    targetRelease: String!
+    assignedTo: String!
+    pmScore: Int!
+    keywords: [String]!
+    # Determined from the externals
+	customerCase: Boolean
+
+*/
+
+// Should this return the string format?
+func (r *BugResolver) DateStamp() string {
+
+	return r.bug.DateStamp.Format("2006-01-02")
+}
+
+func (r *BugResolver) Id() int32 {
+	return int32(r.bug.Id)
+}
+
+func (r *BugResolver) Component() string {
+	// may need to cast to string?
+	return string(r.bug.Component)
+}
+
+func (r *BugResolver) Status() string {
+	return r.bug.Status
+}
+
+func (r *BugResolver) Summary() string {
+	return r.bug.Summary
+}
+
+func (r *BugResolver) TargetRelease() string {
+	// may need to cast to string?
+	return string(r.bug.TargetRelease)
+}
+
+func (r *BugResolver) AssignedTo() string {
+	return r.bug.AssignedTo
+}
+
+func (r *BugResolver) Keywords() []string {
+	return r.bug.Keywords
+}
+
+func (r *BugResolver) PmScore() int32 {
+	return int32(r.bug.PmScore)
+}
+
+// Types for unmarshaling Externals
+type t struct {
+	Description string `json:"description"`
+}
+type external struct {
+	Type t `json:"type"`
+}
+
+// CustomerCase looks through the external links for support cases
+func (r *BugResolver) CustomerCase() bool {
+	// TODO - Should this be done while reading from the db?
+	// We couldn't use bugzilla.bug then...but is that a big deal?
+
+	// Unmarshal the json with the only field we care about
+	var e []external
+	err := json.Unmarshal(r.bug.Externals, &e)
+	if err != nil {
+		log.Printf("Unable to unmarshal externals json: %v\n", err)
+		return false
+	}
+
+	for _, ex := range e {
+		if ex.Type.Description == "Red Hat Customer Portal" {
```

Comment:
I though there was another field that was more succinct.

Reply:
Looking at it, I think the options are:

    type->>description == Red Hat Customer Portal
    type->>id == 60
    ext_bz_id == 60


So I guess ext_bz_id is a layer up which saves us a bit of digging?